Rocm build test rework v2

estewart08 opened the pull request:
The goal is to have amd-staging closer resemble aomp-dev scripts so we can have proper code flow from aomp-dev into amd-staging. This will also let us control testing that is dependent on llvm code changes like those in smoke-limbo.

estewart08 committed:
[build] - First group of build scripts for updated ROCm build/test

estewart08 committed:
Additional build updates for amd-staging support

estewart08 committed:
Additional build updates for amd-staging support - build_offload.sh

estewart08 committed:
Make use of SANITIZER env variable used by ROCm to not rebuild normal libs for asan

estewart08 committed:
Move AOMP_LEGACY_OPENMP after AOMP_STANDALONE_BUILD is set in aomp_common_vars

ampandey-AMD commented on bin/build_offload.sh:
```diff
@@ -117,8 +116,12 @@ if [ "$1" != "nocmake" ] && [ "$1" != "install" ] ; then
+   if [ "$AOMP_STANDALONE_BUILD" == 1 ]; then
+     MYCMAKEOPTS="$COMMON_CMAKE_OPTS -DCMAKE_PREFIX_PATH=$ROCM_CMAKECONFIG_PATH;$AOMP_INSTALL_DIR/lib/cmake -DCMAKE_BUILD_TYPE=Release $AOMP_ORIGIN_RPATH"
+   else
+     MYCMAKEOPTS="$COMMON_CMAKE_OPTS -DCMAKE_PREFIX_PATH=$ROCM_CMAKECONFIG_PATH;$INSTALL_PREFIX/lib/cmake -DCMAKE_BUILD_TYPE=Release $OPENMP_EXTRAS_ORIGIN_RPATH"
```
One concern here does AOMP_INSTALL_DIR and INSTALL_PREFIX expands to same thing for both standalone/non-standalone build.

If same then we can do this below.

```MYCMAKEOPTS``` here only has two  different variable i.e(OPENMP_EXTRAS_ORIGIN_RPATH & INSTALL_PREFIX) in it for non-aomp-standalone build when compared to standalone-build in line 120. can we assign common values for MYCMAKEOPTS only once and just use

```MYCMAKEOPTS="$COMMON_CMAKE_OPTS -DCMAKE_PREFIX_PATH=$ROCM_CMAKECONFIG_PATH;$AOMP_INSTALL_DIR/lib/cmake -DCMAKE_BUILD_TYPE=Release"```

for standalone 

```MYCMAKEOPTS="$MYCMAKEOPTS $AOMP_ORIGIN_RPATH```

for non-standalone
```MYCMAKEOPTS="$MYCMAKEOPTS $OPENMP_EXTRAS_ORIGIN_RPATH"```

estewart08 replied:
For `AOMP_STANDALONE_BUILD=0`, `AOMP_INSTALL_DIR` will be `/opt/rocm/lib/llvm` and `INSTALL_PREFIX` will be `/opt/rocm`. `INSTALL_PREFIX` is not used for `AOMP_STANDALONE_BUILD=1`.

So I want to keep `AOMP_INSTALL_DIR`  only for AOMP builds which uses `AOMP_STANDALONE_BUILD=1`.

ampandey-AMD commented on bin/build_offload.sh:
```diff
@@ -131,7 +134,11 @@ if [ "$1" != "nocmake" ] && [ "$1" != "install" ] ; then
       if [ "$AOMP_BUILD_SANITIZER" == 1 ]; then
-        ASAN_CMAKE_OPTS="$COMMON_CMAKE_OPTS -DCMAKE_PREFIX_PATH=$ROCM_CMAKECONFIG_PATH;$AOMP_INSTALL_DIR/lib/asan/cmake -DSANITIZER_AMDGPU=1 -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_PER_TARGET_RUNTIME_DIR=OFF $AOMP_ASAN_ORIGIN_RPATH"
+        if [ "$AOMP_STANDALONE_BUILD" == 1 ]; then
+          ASAN_CMAKE_OPTS="$COMMON_CMAKE_OPTS -DCMAKE_PREFIX_PATH=$ROCM_CMAKECONFIG_PATH;$AOMP_INSTALL_DIR/lib/asan/cmake -DSANITIZER_AMDGPU=1 -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_PER_TARGET_RUNTIME_DIR=OFF $AOMP_ASAN_ORIGIN_RPATH"
```
Same for ASAN_CMAKE_OPTS also we can apply comment https://github.com/ROCm/aomp/pull/917/files#r1593473768.

One less concern here ```ROCM_CMAKECONFIG_PATH``` is set up by build_openmp_extras.sh script explicitly. So what I am seeing here for non-standalone builds duplicate values in ```CMAKE_PREFIX_PATH```

estewart08 replied:
`AOMP_STANDALONE_BUILD=1` does not use `ROCM_CMAKECONFIG_PATH` so that can be removed.

`AOMP_STANDALONE_BUILD=0` does use it and then we can remove `$INSTALL_PREFIX/lib/asan/cmake` as it does duplicate the path.

estewart08 committed:
Add include path for asan in aomp_common_vars

estewart08 committed:
Adjust component list back to normal for non-standalone build.

estewart08 committed:
Only add offload component if offload cmake is detected for non-standalone

estewart08 committed:
Fix CMAKE_PREFIX_PATH for openmp non-standalone build

estewart08 committed:
Remove asan prefix path from debug only build

estewart08 committed:
Fix typo with AOMP_INSTALL_DIR

estewart08 committed:
Fix CMAKE_PREFIX_PATH for openmp/offload

estewart08 committed:
Only build non-asan libs when SANITIZER != 1

Updates for build_flang.sh, build_pgmath.sh,
and build_flang_legacy.sh. For the normal AOMP build,
which uses AOMP_STANDALONE_BUILD=1 both instrumented and
non-instrumented libraries will be built and installed.
